yarn-project/pxe/src/pxe_service/pxe_service.ts

```diff
@@ -170,27 +170,36 @@ export class PXEService implements PXE {
     return artifact && getContractClassFromArtifact(artifact);
   }
 
-  public async registerAccount(privKey: GrumpkinPrivateKey, partialAddress: PartialAddress): Promise<CompleteAddress> {
-    const completeAddress = CompleteAddress.fromPrivateKeyAndPartialAddress(privKey, partialAddress);
-    const wasAdded = await this.db.addCompleteAddress(completeAddress);
-    if (wasAdded) {
-      const pubKey = await this.keyStore.addAccount(privKey);
-      this.synchronizer.addAccount(pubKey, this.keyStore, this.config.l2StartingBlock);
+  public async registerAccount(secretKey: Fr, partialAddress: PartialAddress): Promise<CompleteAddress> {
```

Comment:
Feels so wrong that it is taking the `secretKey` here. (Also wrong with the privKey).

Reply:
I agree. Created a subtask for this in the [final keys cleanup issue](https://github.com/AztecProtocol/aztec-packages/issues/5726).